### PR TITLE
Miscellaneous chores around cross-compilation

### PR DESCRIPTION
**Pull request by lluiscampos**
Follows from:
* https://github.com/mendersoftware/mender-dist-packages/pull/366

### CODE REVIEW

**Comment by lluiscampos** [comment]
Pinging @kacf mainly FYI (see last commit). Otherwise Ole can do the review alone :mage:

**Review by kacf — APPROVED** [review]
Looks good to me!

File: mender-deb-package [hunk 1]
```diff
@@ -260,7 +260,8 @@ EOF
               CXXFLAGS=" -Wno-psabi"                         \
               GOARCH=arm                                     \
               GOARM=6                                        \
-              dpkg-buildpackage -aarmhf                      \
+              dpkg-buildpackage                              \
+              --target-arch armhf                            \
```
Comment (oleorhagen) [inline mender-deb-package:269]: :heart:

File: mender-deb-package [hunk 2]
```diff
@@ -224,14 +224,19 @@ EOF
...
 
-
-  if ! [ "$OS_DISTRO" = "debian" -a "ARCH" = "armhf" ]; then
-    mk-build-deps --install \
-                  --build-dep \
-                  --tool='apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --yes' \
-                  --host-arch ${ARCH} \
-                  debian/control
-  fi
+  # See MEN-6794 for more details.
```
Comment (oleorhagen) [inline mender-deb-package:227]: Nice writeup!

**Review by oleorhagen — APPROVED** [review]

**Pull request merged**